### PR TITLE
Migrate documentation

### PR DESCRIPTION
**Pull request by hackabletype**
As per issue #2, this PR migrates documentation from https://github.com/dexidp/dex#documentation into this repo.

This time with out all the extra commits.

This is a first pass at bringing in the Documentation. 
* [x] I'm not sure exactly how we want to organize this yet, but I'm pretty sure it's not how we have it on the sidebar in this preview.
  - edit: @celestehorgan has provided code to sort this issue.
* [ ] ~I've also not changed the home page, or the docs index page yet as I'm not exactly sure what should go in those spaces yet either.~ These can be separate issues from this PR.

### CODE REVIEW

**Comment by hackabletype** [comment]
https://deploy-preview-8--dexidp.netlify.app/

**Comment by hackabletype** [comment]
I think we should highlight the docs/proposal section in its own top level menu item

**Comment by hackabletype** [comment]
@celestehorgan could you point me in the direction of how to build some hierarchy into the documentation list (https://deploy-preview-8--dexidp.netlify.app/docs/)

I'd like to have the pages in the `connector` folder be set off from the others, or maybe we could create a section for each subfolder.

**Comment by hackabletype** [comment]
Do we intend to keep multiple versions of documentation around (as per https://deploy-preview-8--dexidp.netlify.app/docs/v2)? If so, we'll probably want to set up versioning right out of the gate.

**Comment by hackabletype** [comment]
@celestehorgan thx for the documentation list fix

File: content/docs/_index.md
Comment (celestehorgan) [inline content/docs/_index.md:2]: Comment: It would be good to provide a _general_ blurb on "what is Dex?" on this page.

cc @justaugustus, please tag other maintainers as you see fit.

File: content/docs/_index.md
Comment (celestehorgan) [inline content/docs/_index.md:3]: All of these `description` tags need to be made more meaningful.

File: content/docs/_index.md
Comment (celestehorgan) [reply content/docs/_index.md:2]: Comment #2: 

Please work with the maintainers to decide on the correct order for pages in these sections.

File: content/docs/api.md
Comment (celestehorgan) [inline content/docs/api.md:10]: Maintainers: Which is the correct capitalization of Dex – lowercase (at all times) or uppercase?

@hackabletype: You don't need to edit all the body content for capitalization in this PR, but please ensure that the titles (and thus menu) are properly capitalized.

File: content/docs/connectors/atlassian-crowd.md
Comment (celestehorgan) [inline content/docs/connectors/atlassian-crowd.md:2]:
```suggestion
title: "Authentication through Atlassian Crowd"
linkTitle: "Atlassian Cloud"
```

This should remove the "Auth through..." phrases on links (i.e., menu items.) it might require some modifications to the sidenav menu. feel free to ping me on slack as a follow-up.

Do for all pages in this section.

File: content/docs/proposals/token-revocation.md
Comment (celestehorgan) [inline content/docs/proposals/token-revocation.md:2]:
```suggestion
title: "Proposal: design for revoking refresh tokens."
linkTitle: "Proposal: design for revoking refresh tokens"
```

1) Be mindful of punctuation/capitalization on all title fields (Sentence capitalization, no punctuation at the end of titles)
2) Same comment as the above on "Authenticating through...". Remove redundant words from link titles for all in this section.

**Review by celestehorgan — COMMENTED** [review]
Initial review! 

Let me know when you need more feedback, and thank you for this great first piece of work, @hackabletype!

**Comment by sagikazarmark** [comment]
@celestehorgan we discussed in one of the issues that we migrate the existing content first and improve it later. The documentation needs a **lot** of improvement in terms of both structure and content. But I guess we can do minor improvements in the first version. Just thought I would mention what we decided earlier.

**Comment by hackabletype** [comment]
To @sagikazarmark's comment, here's the issue where we had some of those discussions: https://github.com/dexidp/dex/issues/1761

**Comment by celestehorgan** [comment]
@sagikazarmark Sounds good to me.

Feel free to disregard any comments that fall under "improve later", @hackabletype, and move this along :)

**Comment by hackabletype** [comment]
@celestehorgan & @sagikazarmark, I've added a commit that adds `linkTitles` to the files under the connectors and proposals folders as I think it adds a nice bit of example. I think we can do the rest of the files as an exercise in improvement.

**Comment by hackabletype** [comment]
The "home" and "docs index" pages are still an open question, but those too can be made into individual issues.

File: content/docs/api.md
Comment (sagikazarmark) [reply content/docs/api.md:10]: Personally, I prefer Dex and tried to use it everywhere. Prior to me joining the project, both was used.

File: content/docs/connectors/atlassian-crowd.md
Comment (sagikazarmark) [reply content/docs/connectors/atlassian-crowd.md]: 👍

**Comment by sagikazarmark** [comment]
"Home" and "Documentation" sound good to me, until we find better names.

File: content/docs/api.md
Comment (hackabletype) [reply content/docs/api.md:10]: I've gone through and made sure that in the titles "Dex" is capitalized as per @sagikazarmark's suggestion.

File: content/docs/proposals/token-revocation.md
Comment (hackabletype) [reply content/docs/proposals/token-revocation.md]: I've gone through and corrected all the title casing throughout, and have removed the redundant words from the linkTitles where they exist.

File: content/docs/_index.md
Comment (hackabletype) [reply content/docs/_index.md:3]: I agree, but think this work can be done on a separate issue.

File: content/docs/_index.md
Comment (hackabletype) [reply content/docs/_index.md:2]: I agree with both these comments, and would like to break them into separate issues to be done after this initial PR.

**Comment by sagikazarmark** [comment]
What do you think should be the strategy for migrating additional changes to the docs until we publish the site?

There are a few PRs with docs changes I'm about to merge.

**Comment by hackabletype** [comment]
@sagikazarmark I'm hoping this PR can be merged soon. I think it's in a place now where we can merge it in as a starting point. Over in the dexidp/dex repo I've got a PR ready that'll remove the docs and point people here. 

That said, if it does take some time, then I think you should make your PR as is, I'll update docs here to keep things in sync. If you do, maybe you could link that PR here just so I know which one will be updating docs?

**Comment by sagikazarmark** [comment]
@hackabletype this is the only PR so far: https://github.com/dexidp/dex/pull/1688

Thanks!

Do you need anything from me to proceed with this PR?

**Comment by hackabletype** [comment]
> @hackabletype this is the only PR so far: [dexidp/dex#1688](https://github.com/dexidp/dex/pull/1688)

Thanks, I'll take a look.

> Do you need anything from me to proceed with this PR?

It just needs an approving review, I've asked one of the original reviewers to re-review it after I made some changes.

**Comment by celestehorgan** [comment]
@hackabletype 

As an initial migration this looks good. Some things that you do in follow-up PRs:

* The content you migrated into the site's index will likely need to be moved into a Getting Started page in the docs proper in future.  We tend to reserve the index pages for marketing purposes on these sites.
* The logo size will need adjusting
* The /docs `_index.md` will need content/a strategy too

**Comment by hackabletype** [comment]
@sagikazarmark, @justaugustus, now that @celestehorgan has reviewed again after the last set of changes, all we need now is an approving review.

Once this PR is approved I'll create some new issues related to the PR feedback we've received here and I'll get the https://github.com/dexidp/dex/pull/1810 PR in so the docs only live in one place.

The https://dexidp.io address is correctly set, and is pointing at the Netlify app now.

**Review by sagikazarmark — APPROVED** [review]

**Comment by sagikazarmark** [comment]
Thanks a lot @hackabletype ! I'm really excited to finally have a proper website for the project.

**Comment by hackabletype** [comment]
You're welcome! And I am too :)

**Pull request merged**
